gensim/similarity_learning/drmm_tks.py

```diff
@@ -0,0 +1,111 @@
+import keras.backend as K
+from keras.models import Model
+from keras.layers import Input, Embedding, Dot, Dense, Lambda, Reshape, Dropout
+from keras.activations import softmax
+
+
+class DRMM_TKS:
+    """This is a variant version of DRMM, which applied topk pooling in the matching matrix.
+    It has the following steps:
+    1. embed queries into embedding vector named 'q_embed' and 'd_embed' respectively
+    2. computing 'q_embed' and 'd_embed' with element-wise multiplication
+    3. computing output of upper layer with dense layer operation
+    4. take softmax operation on the output of this layer named 'g' and find the k largest entries named 'mm_k'.
+    5. input 'mm_k' into hidden layers, with specified length of layers and activation function
+    6. compute 'g' and 'mm_k' with element-wise multiplication.
+
+    # Returns
+        Score list between queries and documents.
+    """
+
+    def __init__(self, embedding, embed_dim, vocab_size, embed_trainable=False, target_mode='ranking',
+                 topk=20, dropout_rate=0., text_maxlen=40, hidden_sizes=[5, 1]):
+        """Initializes the model
+        Parameters:
+        ----------
+        embedding: numpy array matrix
+            A numpy array matrix which has the embeddings extracted from a pretrained
+            word embedding like Stanford Glove
+            This is fed to the Embedding Layer which then outputs the word embedding
+
+        embed_dim: int
+            The dimension of the above mentioned vectors
+
+        vocab_size: int
+            The number of unique words in the corpus
+
+        embed_trainable: boolean
+            Whether the embeddings should be trained
+            if True, the embeddings are trianed
+
+        target_mode: 'training', 'ranking' or 'classification'
+            Indicates the mode in which the model will be used and thus changes the topology
+
+        topk: int
+            Used for topk pooling in the matching matrix
+
+        dropout_rate: float between 0 and 1
+            The probability of making a neuron dead
+            Used for regularization
+
+        text_maxlen: int
+            The maximum possible length of a sentence
+            used for deiciding matrix dimensions
+
+        hidden_sizes: list of ints
```

Comment:
what's size of list? That's non-obviosly

Reply:
I have updated the docstring to
```
hidden_sizes: list of ints
    The list of hidden sizes for the fully connected layers connected to the matching matrix
    For example
        hidden_sizes = [10, 20, 30]
    will add 3 fully connected layers of 10, 20 and 30 hidden neurons
```